### PR TITLE
Edit and redact chat message components

### PR DESCRIPTION
**Pull request by bradmanAWS**
**Description of changes:**
This commit refactors the chat bubble component to allow for editing messages and redacting messages. 

I also removed exports and stories for unreleased chat components and removed the createChatBubblesList function since this logic removes flexibility and should live on the application side.

**Testing**
1. Have you successfully run `npm run build:release` locally?
Yes
2. How did you test these changes?
Storybook and used locally in a demo app
3. If you made changes to the component library, have you provided corresponding documentation changes?
Yes
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/components/ui/Chat/ChatBubble/ChatBubbleContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { StyledChatBubbleContainer, StyledChatBubbleInfo } from './Styled';
+import PopOver from '../../PopOver';
+import { Dots } from '../../icons';
+import { formatTime } from '../../Utilities';
+
+export type Message = {
+  /* The displayed text of the message sent. */
+  content: string;
+  /* The unique identifier of the message. */
+  messageId: string;
```
Comment (p-foucht) [inline src/components/ui/Chat/ChatBubble/ChatBubbleContainer.tsx:16]: Why is this prop necessary? It looks like there's a few props that aren't being used in some of these components - I would be careful about tying the components too closely to an SDK data structure, especially in required props

File: src/components/ui/Chat/ChatBubble/ChatBubble.stories.tsx [hunk 2]
```diff
@@ -1,37 +1,184 @@
...
+        senderName: i%4 === 0 ? 'Alice' : 'Bob',
+        senderId: i%4 === 0 ? 'abc': 'def'
+      }
+      list.push(message);
+    }
+    return list
+  }
+
+  const theme = useTheme();
+  const bgd = theme.chatBubble.container.bgd;
```
Comment (p-foucht) [inline src/components/ui/Chat/ChatBubble/ChatBubble.stories.tsx:63]: I don't think this value is used

File: src/components/ui/Chat/ChatBubble/ChatBubble.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  /* Determines styling for outgoing and incoming messages. */
+  variant: MessageVariant;
+  /* The name of the user that sent the message. */
+  senderName: string;
+  /* The time the message was last edited */
+  lastUpdatedTimestamp?: string;
+  /* The text content of the message. */
+  content: string;
+  /* Determines if the name should be shown or not. */
+  showName?: boolean;
```
[the remark this answers is not in the document]
Comment (bradmanAWS) [reply src/components/ui/Chat/ChatBubble/ChatBubble.tsx:21]: I'm not 100% sure I understand the question, but there are definitely stylistic reasons why we might not want to display the name even though it's provided.

File: src/components/ui/Chat/ChatBubble/ChatBubbleContainer.tsx (hunk 1 above)
Comment (bradmanAWS) [reply src/components/ui/Chat/ChatBubble/ChatBubbleContainer.tsx:16]: Yep, this was supposed to be removed.

File: src/components/ui/Chat/ChatBubble/ChatBubble.stories.tsx (hunk 2 above)
Comment (bradmanAWS) [reply src/components/ui/Chat/ChatBubble/ChatBubble.stories.tsx:63]: fixed

File: src/components/ui/Chat/ChannelList/ChannelList.stories.tsx [hunk 4]
```diff
@@ -6,18 +6,18 @@ import React, { useState } from 'react';
...
-export default {
-  title: 'UI Components/Chat',
-  parameters: {
-    docs: {
-      page: ChannelListDocs.parameters.docs.page().props.children.type
-    }
-  },
-  component: ChannelList
-};
+// export default {
```
Comment (nicklera1) [inline src/components/ui/Chat/ChannelList/ChannelList.stories.tsx:12]: all of this is commented out!

File: src/components/ui/Chat/ChatBubble/ChatBubble.stories.tsx [hunk 5]
```diff
@@ -1,37 +1,184 @@
...
-  title: 'UI Components/Chat',
-  parameters: {
-    docs: {
-      page: ChatBubbleDocs.parameters.docs.page().props.children.type
-    }
-  },
-  component: ChatBubble
-};
+// export default {
+//   title: 'UI Components/Chat',
```
Comment (nicklera1) [inline src/components/ui/Chat/ChatBubble/ChatBubble.stories.tsx:16]: Commented out! Also make sure Title is unique for each Chat Storie.

File: src/components/ui/Chat/ChannelList/ChannelList.stories.tsx (hunk 4 above)
Comment (bradmanAWS) [reply src/components/ui/Chat/ChannelList/ChannelList.stories.tsx:12]: Yeah this is intentional. We don't want these stories published to storybook before the components are ready for release.

File: src/components/ui/Chat/DateHeader/DateHeader.stories.tsx [hunk 6]
```diff
@@ -4,18 +4,18 @@
...
-export default {
-  title: 'UI Components/Chat/DateHeader',
-  parameters: {
-    docs: {
-      page: DateHeaderDocs.parameters.docs.page().props.children.type
-    }
-  },
-  component: DateHeader
-};
+// export default {
```
Comment (nicklera1) [inline src/components/ui/Chat/DateHeader/DateHeader.stories.tsx:10]: uncomment

File: src/components/ui/Chat/ChatBubble/ChatBubble.stories.tsx (hunk 5 above)
Comment (bradmanAWS) [reply src/components/ui/Chat/ChatBubble/ChatBubble.stories.tsx:16]: I'm not sure about this one. We get a yellow console warning for this, I know, but this way storybook nicely organizes the stories into a folder structure that resembles the code's folder structure.

File: src/components/ui/Chat/InfiniteList/InfiniteList.stories.tsx [hunk 7]
```diff
@@ -1,25 +1,21 @@
...
-  title: 'UI Components/Chat/ChatList',
-  parameters: {
-    docs: {
-      page: InfiniteListDocs.parameters.docs.page().props.children.type
-    }
-  },
-  component: InfiniteList
-};
+// export default {
+//   title: 'UI Components/Chat/InfiniteList',
```
Comment (nicklera1) [inline src/components/ui/Chat/InfiniteList/InfiniteList.stories.tsx:11]: uncomments

File: src/components/ui/Chat/ChatBubble/ChatBubble.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  /* Adds the bubble tail style to a message. */
+  showTail?: boolean;
+  /* Determines if the message has been removed after being sent. */
+  redacted?: boolean;
+  /* The text to be shown in place of the message content when the message has been redacted. */
+  redactedText?: string;
```
[the remark this answers is not in the document]
Comment (bradmanAWS) [reply src/components/ui/Chat/ChatBubble/ChatBubble.tsx:27]: Got it. Updating.

**Review by nicklera1 — APPROVED** [review]

**Pull request merged**